Send notification when signature threshold reached

armandfardeau opened the pull request:
#### :tophat: What? Why?
This PR will add a notification that will be sent authors, promoting committee members, followers of the initiative and admins to receive a notification when an initiative reaches the support threshold.

linked to : https://meta.decidim.org/processes/roadmap/f/122/proposals/15280

#### :clipboard: Subtasks
- [x] Add `CHANGELOG` entry
- [x] Add tests

virgile-dev commented:
@carolromero this need your approval to be merged :)

virgile-dev commented:
@armandfardeau looks like we didn't understand each other on this one.
It's the admin who click on the button to accept an initiative so it doesn't make sense to send him and email when accepting an initiative. Same for rejecting.

It does for authors, followers, promotting committee members though. So maybe we can slightly revamp this to exclude admin and include authors, followers, promotting committee members.

What I wanted to do with this PR is to send a notification to the admin when the initiative reaches the 100% milestone. (see this [PR](https://github.com/decidim/decidim/pull/3341/files) for ref)

virgile-dev commented:
@carolromero is this one okay with you ? I see that @andreslucena had a look.

Leusev commented on decidim-initiatives/app/commands/decidim/initiatives/vote_initiative.rb:
```diff
@@ -92,6 +93,21 @@ def notify_percentage_change(before, after)
+        )
+      end
+
+      def organization_admins
```
As I saw in last paragraph of this @virgile-dev comment, he asked to send a notification to the admin when threshold was reached:
![image](https://user-images.githubusercontent.com/9702463/82569287-51e06100-9b80-11ea-90c7-f0167362c9db.png)
but as I saw there:
![image](https://user-images.githubusercontent.com/9702463/82569406-7dfbe200-9b80-11ea-8de9-4849df4da893.png)
he makes a changelog PR description's change suggestion as if the notification had to be sent to all them

As code shows the notification only will be sent to organization's admins, could you check/clarify this please @armandfardeau @virgile-dev ?

Thanks in advance!

virgile-dev replied:
@armandfardeau I was under the impressions we'd notify the author, the promoting committee and the followers. Can you make the necessary changes please ?

armandfardeau replied:
For sure, sorry for the mess.

As far as I understood, we already send a notification to authors, followers, and committee members. 
The notification is included is `notify_percentage_change`.

In order to customize the notifications, a new method has been added `notify_support_threshold_reached` which notifies only admins.

Let me know if issues are still hanging around

Leusev replied:
So method `notify_percentage_change` is already sending a notification to authors, followers and commitee members for percentage changes between 25, 50, 75, 100 ?
It looks like as author wouldn't be notified and committee members would be only if they already are initiative's followers, is this right?

![image](https://user-images.githubusercontent.com/9702463/82656467-1d25e580-9c24-11ea-9650-5bf97e286085.png)

The new `notify_support_threshold_reached` method only will send a notification to admin when 100% is reached.

Could you confirm that this is how it should be please?
Thanks in advance

armandfardeau replied:
@Leusev Yes, this way we can customize the admin notifications

@virgile-dev Mind to double-check?

virgile-dev replied:
Yes it's perfect to have a different one for admins for customization purposes. I approve 👍

andreslucena commented:
> @carolromero is this one okay with you ? I see that @andreslucena had a look 

Hey all, I've actually didn't look at this, it's just that I hate long titles on PR/commits. I've updated the docs with this a few weeks ago, please take a look at https://github.com/decidim/decidim/blob/develop/docs/development_guide.md#git-commit-messages-and-pull-request-titles 

The title "Notify authors, promoting committee members, followers of the initiative and admins to receive a notification when an initiative reaches the 100% of the support threshold" is 171 characters
Hell, even the new title is too long by the guidelines that I've linked (we're on 71 chars, guidelines says 50 chars), but I can live with that for the time being. 

(Still the title is shorter than the title of Garcia Marquez book "Relatos de un naufrago" / "The Story of a Shipwrecked Sailor" which full title is "The Story of a Shipwrecked Sailor: Who Drifted on a Liferaft for Ten Days Without Food or Water, Was Proclaimed a National Hero, Kissed by Beauty Queens, Made Rich Through Publicity, and Then Spurned by the Government and Forgotten for All Time.", 246 chars!!)

armandfardeau commented:
> > @carolromero is this one okay with you ? I see that @andreslucena had a look
> 
> Hey all, I've actually didn't look at this, it's just that I hate long titles on PR/commits. I've updated the docs with this a few weeks ago, please take a look at https://github.com/decidim/decidim/blob/develop/docs/development_guide.md#git-commit-messages-and-pull-request-titles
> 
> The title "Notify authors, promoting committee members, followers of the initiative and admins to receive a notification when an initiative reaches the 100% of the support threshold" is 171 characters
> Hell, even the new title is too long by the guidelines that I've linked (we're on 71 chars, guidelines says 50 chars), but I can live with that for the time being.
> 
> (Still the title is shorter than the title of Garcia Marquez book "Relatos de un naufrago" / "The Story of a Shipwrecked Sailor" which full title is "The Story of a Shipwrecked Sailor: Who Drifted on a Liferaft for Ten Days Without Food or Water, Was Proclaimed a National Hero, Kissed by Beauty Queens, Made Rich Through Publicity, and Then Spurned by the Government and Forgotten for All Time.", 246 chars!!)

I was not aware of these new guidelines, thank you for the clarification. I've updated this P.R., let me know if anything is still out of order.

virgile-dev commented:
@carolromero thanks for accepting the proposal on meta decidim. Does this mean that @Leusev can merge the PR ?

carolromero commented:
@virgile-dev yep, it's ok on my side

armandfardeau commented:
@Leusev GH actions had a malfunction, can you restart them on your side? Everything should be green by now

Leusev commented:
Hi @armandfardeau 
for sure, I've re-launched "[CI] admin / Tests" again
I know it is repetitive, but it seems that the changelog has conflict again
Thanks!

armandfardeau commented:
> Hi @armandfardeau
> for sure, I've re-launched "[CI] admin / Tests" again
> I know it is repetitive, but it seems that the changelog has conflict again
> Thanks!

I should have done my chores a long time ago ;)

virgile-dev commented:
@Leusev could you check if we can merge this now ? Thanks !
